### ISSUE TITLE
[General Issue]  Getting test results for my data

### ISSUE THREAD

**Issue by Yasminbr** [PaddlePaddle/PaddleSeg#2222]
Hi, 

After I trained my UNET I would like to get the results of Dice, Iou,Kappa etc. for the test data. But unfortunately I can not achieve this. 
The model is doing validation while training but i want him to check for the test data after all the iterations are finishing.
This is my yml file. 

`batch_size: 6
iters: 6000
num_workers: 2
train_dataset:
  type: Dataset
  dataset_root: /home/ARO.local/yasminbr/Projects/June/
  num_classes: 2
  train_path: /home/ARO.local/yasminbr/Projects/June/train.txt
  transforms:
    - type: Resize
      target_size: [512, 512]
    - type: Normalize
      mean: [0.3644, 0.4067, 0.2521]
      std: [0.2360, 0.1924, 0.1574]
  mode: train

val_dataset:
  type: Dataset
  dataset_root: /home/ARO.local/yasminbr/Projects/June
  num_classes: 2
  val_path: /home/ARO.local/yasminbr/Projects/June/val.txt

  transforms:
    - type: Resize
      target_size: [512, 512]
    - type: Normalize
      mean: [0.3644, 0.4067, 0.2521]
      std: [0.2360, 0.1924, 0.1574]
  mode: val

test_dataset:
  type: Dataset
  dataset_root: /home/ARO.local/yasminbr/Projects/June/
  num_classes: 2
  test_path: /home/ARO.local/yasminbr/Projects/June/test.txt
  mode: test

optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 4.0e-5

lr_scheduler:
  type: PolynomialDecay
  learning_rate: 0.001
  end_lr: 0
  power: 0.2

loss:
  types:
    - type: DiceLoss
  coef: [1]
  ignore_index: 0

model:
  type: UNet
  num_classes: 2
  use_deconv: False
  pretrained: /home/ARO.local/yasminbr/Projects/June/PreTrained/UnetCityPassmodel.pdparams
`

**Comment by shiyutang** [comment]
Normally, the test data do not have labels. Therefore you can not get various metric results on it like kappa, dice and etc. However, you can use [predict.py](https://github.com/PaddlePaddle/PaddleSeg/blob/release%2F2.5/predict.py) to predict and visualize the images.

**Comment by Yasminbr** [comment]
so what is the "Test mode" for? 

test_dataset:
type: Dataset
dataset_root: /home/ARO.local/yasminbr/Projects/June/
num_classes: 2
test_path: /home/ARO.local/yasminbr/Projects/June/test.txt
mode: test

**Comment by shiyutang** [comment]
Test dataset is used to test the performance of your model, see how it react on data it never sees.

**Comment by Yasminbr** [comment]
> Test dataset is used to test the performance of your model, see how it react on data it never sees.

I know the meaning of the test data but when running this command I don't get any feedback from the net.

**Comment by shiyutang** [comment]
If you are evaluating with multiple GPUs, the result will be produced after some time without any other information. You can see the GPU is occupied but there is no other information. Please wait patiently for the result to come back.
